### ISSUE TITLE
CrashLoopBackOff on GKE

### ISSUE THREAD

**Issue by PascalBourdier** [loft-sh/vcluster#1405]
### What happened?

`CrashLoopBackOff` on pod

### What did you expect to happen?

the pod should run without error

### How can we reproduce it (as minimally and precisely as possible)?

```
kubectl create namespace vcluster-my-vcluster
helm template my-vcluster vcluster --repo https://charts.loft.sh -n vcluster-my-vcluster | kubectl apply -f -
```

### Anything else we need to know?

```
$ k -n vcluster-my-vcluster logs my-vcluster-0
Defaulted container "syncer" out of: syncer, vcluster (init)
2023-12-11 12:50:27	INFO	k3s/k3s.go:71	Starting k3s	{"component": "vcluster", "args": "/k3s-binary/k3s server --write-kubeconfig=/data/k3s-config/kube-config.yaml --data-dir=/data --disable=traefik,servicelb,metrics-server,local-storage,coredns --disable-network-policy --disable-agent --disable-cloud-controller --egress-selector-mode=disabled --flannel-backend=none --kube-apiserver-arg=bind-address=127.0.0.1 --disable-scheduler --kube-controller-manager-arg=controllers=*,-nodeipam,-nodelifecycle,-persistentvolume-binder,-attachdetach,-persistentvolume-expander,-cloud-node-lifecycle,-ttl --kube-apiserver-arg=endpoint-reconciler-type=none --service-cidr 10.32.0.0/16 --token ovjotzqehcxrajxtnodutbxmtzgaxxdcxwrizidrfqcymlkbuvzgswkaszuxqhke"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1faffe5]

goroutine 1 [running]:
github.com/loft-sh/vcluster/pkg/telemetry.(*DefaultCollector).RecordError(0xc000636c00, {0x309d3f0, 0xc0009cc0c0}, {0x2aca930, 0x5}, {0x306e340, 0xc000cccc40})
	github.com/loft-sh/vcluster/pkg/telemetry/collect.go:234 +0x1a5
github.com/loft-sh/vcluster/cmd/vcluster/cmd.NewStartCommand.func1.1()
	github.com/loft-sh/vcluster/cmd/vcluster/cmd/start.go:61 +0x15b
panic({0x266d620?, 0x48fc850?})
	runtime/panic.go:914 +0x21f
github.com/loft-sh/vcluster/pkg/telemetry.(*DefaultCollector).RecordStart(0xc000636c00, {0x309d3f0, 0xc0009cc0c0})
	github.com/loft-sh/vcluster/pkg/telemetry/collect.go:198 +0xc5
github.com/loft-sh/vcluster/pkg/setup.Initialize({0x309d3f0, 0xc0009cc0c0}, {0x30cb770?, 0xc000571ba0?}, {0x30cb770?, 0xc000571ba0?}, {0xc0000565e8?, 0x14?}, {0xc0000565e8, 0x14}, ...)
	github.com/loft-sh/vcluster/pkg/setup/initialize.go:64 +0x262
github.com/loft-sh/vcluster/cmd/vcluster/cmd.ExecuteStart({0x309d3f0, 0xc0009cc0c0}, 0xc000990e00)
	github.com/loft-sh/vcluster/cmd/vcluster/cmd/start.go:112 +0x22b
github.com/loft-sh/vcluster/cmd/vcluster/cmd.NewStartCommand.func1(0xc000004900, {0xc0005e6540?, 0x4?, 0x2ac9c14?})
	github.com/loft-sh/vcluster/cmd/vcluster/cmd/start.go:69 +0xd8
github.com/spf13/cobra.(*Command).execute(0xc000004900, {0xc0005e6300, 0x6, 0x6})
	github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004600)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(0x309d3b8?, {0x309d3f0?, 0xc0009cc0c0?})
	github.com/spf13/cobra@v1.8.0/command.go:1032 +0x47
main.main()
	github.com/loft-sh/vcluster/cmd/vcluster/main.go:42 +0x1bc
```

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.28.4
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.24.17-gke.200
WARNING: version difference between client (1.28) and server (1.24) exceeds the supported minor version skew of +/-1
```

</details>

### Host cluster Kubernetes distribution

<details>

```
v1.24.17-gke.200
```

</details>

### vlcuster version

<details>

```console
$ vcluster --version
vcluster version 0.18.0
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
v1.24.17-gke.200
```

</details>

### OS and Arch

<details>

for node:
```
OS: Container-Optimized OS from Google
Arch: amd64
```
</details>

**Comment by FabianKramm** [comment]
@PascalBourdier that should be fixed with the latest release v0.18.1, can you check again?

**Comment by PascalBourdier** [comment]
@FabianKramm it's ok now:
```
$ k -n vcluster-my-vcluster get po
NAME                                                   READY   STATUS    RESTARTS   AGE
coredns-68bdd584b4-bjvgd-x-kube-system-x-my-vcluster   1/1     Running   0          21m
my-vcluster-0                                          1/1     Running   0          22m
```
